src/notebook/plugin.ts

```diff
@@ -230,12 +230,15 @@ function addCommands(app: JupyterLab, services: IServiceManager, tracker: Notebo
       }
     }
   });
-  commands.addCommand(CommandIDs.closeAndHalt, {
-    label: 'Close and Halt',
+  commands.addCommand(CommandIDs.closeAndShutdown, {
+    label: 'Close and Shutdown',
     execute: () => {
       let current = tracker.currentWidget;
       if (current) {
-        current.context.changeKernel(null).then(() => { current.dispose(); });
+        app.shell.activateMain(current.id);
+        if (confirm('Are you sure you want to close the notebook?')) {
```

Comment:
Do you mind using our dialog for this?   See https://github.com/jupyterlab/jupyterlab/blob/522bfd89d38ca3cae65a2d83e3f96baf6e421263/src/notebook/trust.ts#L49 for example.